src/components/target/TargetDetails.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import { func, object, array, bool } from 'prop-types';
+import SmilesLogo from 'resources/icons/SmilesLogo';
+import TargetIcon from 'resources/icons/TargetIcon';
+import { deleteTarget, endSelectedTarget } from 'actions/targetActions';
+import MenuHeader from 'components/common/MenuHeader';
+import Modal from 'components/common/Modal';
+import Loading from 'components/common/Loading';
+import { getTopicIcon } from 'utils/helpers';
+
+class TargetDetails extends Component {
+  constructor() {
+    super();
+
+    this.confirmDelete = this.confirmDelete.bind(this);
+    this.closeDeleteModal = this.closeDeleteModal.bind(this);
+  }
+  state = { confirmDelete: false };
+
+  getTopicLabel(topicId) {
```

Comment:
Do you think this function could be reused with the getTopicIcon? Most of the code is the same.